metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -500,9 +631,26 @@ boolean uncleanLeaderElectionEnabledForTopic(String name) {
         return false; // TODO: support configuring unclean leader election.
     }
 
+    Map<String, ConfigEntry> computeEffectiveTopicConfigsWithPendingChange(
+        Map<ConfigResource, Map<String, String>> pendingConfigData
+    ) {
+        Map<String, String> pendingClusterConfig =
+            pendingConfigData.containsKey(DEFAULT_NODE) ? pendingConfigData.get(DEFAULT_NODE) : Collections.emptyMap();
+        Map<String, String> pendingControllerConfig =
+            pendingConfigData.containsKey(currentController) ? pendingConfigData.get(currentController) : Collections.emptyMap();
+        return configSchema.resolveEffectiveTopicConfigs(
+            new OrderedConfigResolver(staticConfig),
```

Comment:
Hm.. what happens if the controller has different static configs from the broker?